tests/andromede/test_quota_co2.py

```diff
@@ -0,0 +1,194 @@
+from andromede.expression import literal, param, var
+from andromede.expression.expression import port_field
+from andromede.libs.standard import CONSTANT, TIME_AND_SCENARIO_FREE
+from andromede.model import (
+    Constraint,
+    ModelPort,
+    PortField,
+    PortType,
+    float_parameter,
+    float_variable,
+    model,
+)
+from andromede.model.model import PortFieldDefinition, PortFieldId
+from andromede.simulation import TimeBlock, build_problem, OutputValues
+from andromede.study import (
+    ConstantData,
+    DataBase,
+    Network,
+    Node,
+    PortRef,
+    create_component,
+)
+
+"""
+Power flow port 
+"""
+FLOW_PORT = PortType(id="flow_port", fields=[PortField("F")])
+
+"""
+CO² emmission port
+"""
+EMISSION_PORT = PortType(id="emission_port", fields=[PortField("Q")])
+
+"""
+Simple node that manage flow interconnections.
+"""
+NODE_MODEL = model(
+    id="Noeud",
+    ports=[ModelPort(port_type=FLOW_PORT, port_name="FlowN")],
+    binding_constraints=[
+        Constraint(
+            name="Balance",
+            expression=port_field("FlowN", "F").sum_connections() == literal(0),
+        )
+    ],
+)
+
+"""
+Model of a simple power generator that takes account of CO² emissions related to the production.
+The power production p is bounded between p_min and p_max.
+An emission factor is used to determine the CO² emission according to the production.
+"""
+C02_POWER_MODEL = model(
+    id='CO2 power',
+    parameters=[float_parameter("p_min", CONSTANT),
+                float_parameter("p_max", CONSTANT),
+                float_parameter("cost", CONSTANT),
+                float_parameter("taux_emission", CONSTANT)],
+    variables=[float_variable("p", lower_bound=param("p_min"), upper_bound=param("p_max"))],
+    ports=[ModelPort(port_type=FLOW_PORT, port_name="FlowP"),
+           ModelPort(port_type=EMISSION_PORT, port_name="OutCO2")],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("FlowP", "F"),
+            definition=var("p"),
+        ),
+        PortFieldDefinition(
+            port_field=PortFieldId("OutCO2", "Q"),
+            definition=var("p") * param("taux_emission"),
+        )
+    ],
+    objective_contribution=(param("cost") * var("p")).sum().expec(),
+)
+
+
+"""
+Basic energy consumption model.
+It consume a fixed amount of energy "d" each hour.
+"""
+DEMAND_MODEL = model(
+    id='Demand model',
+    parameters=[float_parameter("d", CONSTANT)],
+    ports=[ModelPort(port_type=FLOW_PORT, port_name="FlowD")],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("FlowD", "F"),
+            definition=-param("d"),
+        )
+    ]
+)
+
+"""
+Model of the CO² quota. 
+It takes a set a CO² emissions as input. It forces the sum of those emissions to be smaller than a predefined quota. 
+"""
+QUOTA_CO2_MODEL = model(
```

Comment:
Maybe it is better that you put your new models in `libs`, probably in a new file, so that we gather all models in the same folder and not everywhere across tests